### PR TITLE
fix linter issues

### PR DESCRIPTION
**Pull request by bennyz**

### CODE REVIEW

File: pkg/controller/map/storage/handler/ocp/handler.go [hunk 1]
```diff
@@ -68,6 +69,7 @@ func (r *Handler) changed(storageClass *ocp.StorageClass) {
 	err := r.List(context.TODO(), &list)
 	if err != nil {
 		err = liberr.Wrap(err)
+		log.Error(err, "failed to list StorageMap CRs")
```
Comment (ahadas) [inline pkg/controller/map/storage/handler/ocp/handler.go:72]: does the linter look for logging along with errors?
@liranr23 ^ related to what we discussed on the PR that fixed the calls to liberr.Wrap

File: pkg/lib/filebacked/list_test.go [hunk 2]
```diff
@@ -192,7 +193,9 @@ func TestList(t *testing.T) {
 }
 
 // Disabled by default.
-func __TestListPerf(t *testing.T) {
+func TestListPerf(t *testing.T) {
+	t.Skip("Skipping performance test")
```
Comment (ahadas) [inline pkg/lib/filebacked/list_test.go:197]: ah that's what you meant :D

File: tests/suit/utils/pod.go [hunk 3]
```diff
@@ -100,14 +98,19 @@ func FindPodByPrefixOnce(clientSet *kubernetes.Clientset, namespace, prefix, lab
...
 		if result != nil {
 			return true, err
 		}
-		// If no result yet, continue polling even if there is an error
 		return false, nil
 	})
+
+	if err != nil {
+		return nil, err
+	}
```
Comment (ahadas) [inline tests/suit/utils/pod.go:112]: what error was fixed by this addition?

Comment (bennyz) [reply tests/suit/utils/pod.go:112]: not sure 😅

File: pkg/controller/map/storage/handler/ocp/handler.go (hunk 1 above)
Comment (ahadas) [reply pkg/controller/map/storage/handler/ocp/handler.go:72]: I see, makes sense

File: pkg/lib/inventory/model/table.go [hunk 4]
```diff
@@ -683,7 +681,11 @@ func (t Table) EnsurePk(md *Definition) {
 			reflect.Int32,
 			reflect.Int64:
 			bfr := new(bytes.Buffer)
-			binary.Write(bfr, binary.BigEndian, f.int)
+			err := binary.Write(bfr, binary.BigEndian, f.int)
+			if err != nil {
+				panic(err)
```
[the remark this answers is not in the document]
Comment (bennyz) [reply pkg/lib/inventory/model/table.go:686]: we don't return anything...

File: pkg/controller/provider/container/ovirt/collector.go [hunk 5]
```diff
@@ -483,7 +493,7 @@ func (r *Collector) apply(changeSet []Updater) (err error) {
 func (r *Collector) listEvent() (list []Event, err error) {
 	eventList := EventList{}
 	codes := []string{}
-	for n, _ := range adapterMap {
+	for n := range adapterMap {
```
[the remark this answers is not in the document]
Comment (bennyz) [reply pkg/controller/provider/container/ovirt/collector.go:496]: it's a map, not a list, so it's the map's key

File: pkg/lib/inventory/model/table.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (ahadas) [reply pkg/lib/inventory/model/table.go:686]: so a log maybe?

File: pkg/controller/provider/container/ovirt/collector.go [hunk 6]
```diff
@@ -432,20 +432,30 @@ func (r *Collector) refresh(ctx *Context) (err error) {
 			"event",
 			event)
 		var changeSet []Updater
-		changeSet, applyErr := r.changeSet(ctx, event)
-		if applyErr == nil {
-			applyErr = r.apply(changeSet)
-			r.log.V(3).Info(
-				"Event applied.",
+		changeSet, err := r.changeSet(ctx, event)
```
Comment (ahadas) [inline pkg/controller/provider/container/ovirt/collector.go:435]:
```suggestion
		changeSet, err = r.changeSet(ctx, event)
```

File: pkg/controller/provider/container/ovirt/collector.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (ahadas) [reply pkg/controller/provider/container/ovirt/collector.go:501]: ah right

File: tests/suit/utils/pod.go [hunk 7]
```diff
@@ -100,14 +98,15 @@ func FindPodByPrefixOnce(clientSet *kubernetes.Clientset, namespace, prefix, lab
...
 	var result *k8sv1.Pod
 	var err error
-	wait.PollImmediate(2*time.Second, podCreateTime, func() (bool, error) {
+
+	err = wait.PollUntilContextCancel(context.TODO(), 2*time.Second, true, func(ctx context.Context) (bool, error) {
 		result, err = findPodByCompFuncOnce(clientSet, namespace, prefix, labelSelector, compFunc)
 		if result != nil {
 			return true, err
 		}
-		// If no result yet, continue polling even if there is an error
```
[the remark this answers is not in the document]
Comment (ahadas) [reply tests/suit/utils/pod.go:108]: you restored it in a different place :)

**Review by ahadas — APPROVED** [review]
on the one hand, it's pretty late in the development cycle of 2.6 but, on the other hand, if we wait with this until we branch out, it may make backports more complex - so let's get it in to the main branch before we branch out 2.6

**Pull request merged**